src/tests/ftest/util/server_utils.py

```diff
@@ -50,6 +57,439 @@ class ServerFailed(Exception):
+        """Start the server through the runner."""
+        self.runner.job.set_config(yamlfile)
+        self.server_clean()
+
+        # Prepare nvme storage in servers
+        if self.runner.job.yaml_params.is_nvme():
+            self.log.info("Performing nvme storage prepare in <format> mode")
+            storage_prepare(self._hosts)
+            self.runner.job.sudo = True
+
+        try:
+            self.run()
+        except CommandFailure as details:
+            self.log.info("<SERVER> Exception occurred: %s", details)
+            # Kill the subprocess, anything that might have started
+            self.kill()
+            raise ServerFailed(
+                "Failed to start server in {} mode.".format(
+                    self.runner.job.mode))
+
+        if self.runner.job.yaml_params.is_nvme() or \
+           self.runner.job.yaml_params.is_scm():
+            # Setup the hostlist to pass to dmg command
+            servers_with_ports = [
+                "{}:{}".format(host, self.runner.job.yaml_params.port)
+                for host in self._hosts]
+
+            self.log.info("Formatting hosts: <%s>", self._hosts)
+            storage_format(self.daosbinpath, ",".join(servers_with_ports))
+            self.runner.job.mode = "postformat"
+            try:
+                self.runner.job.check_subprocess_status(self.runner.process)
+            except CommandFailure as error:
+                self.log.info("Failed to start after format: %s", error)
+
+        return True
+
+    def stop(self):
+        """Stop the server through the runner."""
+        if self.runner.job.yaml_params.is_nvme() or \
+           self.runner.job.yaml_params.is_scm():
+            # Setup the hostlist to pass to dmg command
+            servers_with_ports = [
+                "{}:{}".format(host, self.runner.job.yaml_params.port)
+                for host in self._hosts]
+            storage_reset(
+                self.daosbinpath, ",".join(servers_with_ports), nvme=True)
+
+        self.log.info("Stopping servers")
+        try:
+            self.runner.stop()
+        except CommandFailure as error:
+            raise ServerFailed("Failed to stop servers:{}".format(error))
+
+    def server_clean(self):
+        """Prepare the hosts before starting daos server."""
+        # Kill any doas servers running on the hosts
+        self.kill()
+        # Clean up any files that exist on the hosts
+        self.clean_files()
+
+    def kill(self):
+        """Forcably kill any daos server processes running on hosts.
+
+        Sometimes stop doesn't get everything.  Really whack everything
+        with this.
+
+        """
+        kill_cmds = [
+            "sudo pkill '(daos_server|daos_io_server)' --signal INT",
+            "sleep 5",
+            "pkill '(daos_server|daos_io_server)' --signal KILL",
+        ]
+        # Intentionally ignoring the exit status of the command
+        self.log.info("Killing any server processes")
```

Comment:
Suggestion to improve potential debug (could be added later):
```suggestion
        self.log.info(
            "Killing any server processes on %s",
            str(NodeSet.fromlist(self._hosts)))
```